native/cocos/renderer/gfx-agent/DeviceAgent.cpp

```diff
@@ -284,23 +285,32 @@ void doBufferTextureCopy(const uint8_t *const *buffers, Texture *texture, const
     for (uint32_t i = 0U; i < count; i++) {
         bufferCount += regions[i].texSubres.layerCount;
     }
-    uint32_t totalSize = sizeof(BufferTextureCopy) * count + sizeof(uint8_t *) * bufferCount;
+    
+    auto alignTo = [](size_t size, uint32_t alignment) {
+        return (size + alignment - 1) / alignment * alignment;
+    };
+    
```

Comment:
We might use boost::alignment::align_up
https://www.boost.org/doc/libs/1_79_0/boost/align/align_up.hpp